pkg/os/volume/api.go

```diff
@@ -186,18 +192,24 @@ func (VolumeAPI) ResizeVolume(volumeID string, size int64) error {
 	} else {
 		finalSize = size
 	}
+	klog.V(4).Infof("finalSize for resize of the volume: %s is %d ", volumeID, finalSize)
 
 	currentSize, err := getVolumeSize(volumeID)
 	if err != nil {
 		return fmt.Errorf("error getting the current size of volume (%s) with error (%v)", volumeID, err)
 	}
-
+	klog.V(4).Infof("currentSize of the volume: %s is %d ", volumeID, currentSize)
 	//if the partition's size is already the size we want this is a noop, just return
 	if currentSize >= finalSize {
 		klog.V(2).Infof("Attempted to resize volume %s to a lower size, from currentBytes=%d wantedBytes=%d", volumeID, currentSize, finalSize)
 		return nil
 	}
-
+	// Resize-Partition requires minimum extent size 1 MB
+	// if extent size for resize partition is less than 1 MB, skip expanding volume
+	if finalSize-currentSize < MinimumExpandSize {
+		return status.Errorf(codes.OutOfRange,
+			"minimum extent size is 1 MB. Skip resize for volume %s from currentBytes=%d to wantedBytes=%d ", volumeID, currentSize, finalSize)
+	}
```

Comment:
Does this fixes the issue we observed btw? It sounds like - this will prevent node-expansion if increased size is less than 1MB?